### PR TITLE
Configurable Github/Gitea urls

### PR DESCRIPTION
**Pull request by gerdemann**

### CODE REVIEW

File: src/Playbloom/Satisfy/Webhook/GithubWebhook.php [hunk 1]
```diff
@@ -25,10 +25,18 @@
 
 class GithubWebhook extends AbstractWebhook
 {
-    public function __construct(Manager $manager, EventDispatcherInterface $dispatcher, ?string $secret = null)
+    /** @var array */
```
Comment (ramunasd) [inline src/Playbloom/Satisfy/Webhook/GithubWebhook.php:28]: Please define array shape

File: src/Playbloom/Satisfy/Webhook/GithubWebhook.php [hunk 2]
```diff
@@ -25,10 +25,18 @@
...
+    protected $sourceUrls = ['git_url', 'ssh_url', 'clone_url', 'svn_url'];
+
+    public function __construct(
+        Manager $manager, EventDispatcherInterface $dispatcher, 
+        ?string $secret = null,
+        ?array $sourceUrls = ['git_url', 'ssh_url', 'clone_url', 'svn_url']
```
Comment (ramunasd) [inline src/Playbloom/Satisfy/Webhook/GithubWebhook.php:34]: Can You set this default value directly to property?

File: src/Playbloom/Satisfy/Webhook/GithubWebhook.php [hunk 3]
```diff
@@ -25,10 +25,18 @@
...
+    )
     {
         parent::__construct($manager, $dispatcher);
         $this->secret = $secret;
+        $this->sourceUrls = $sourceUrls;
```
Comment (ramunasd) [inline src/Playbloom/Satisfy/Webhook/GithubWebhook.php:39]: Please use setter for this property as it is optional and set by default.

**Review by ramunasd — CHANGES_REQUESTED** [review]

**Comment by gerdemann** [comment]
@ramunasd Can you look at it again, please? Is that what you meant?

**Comment by ramunasd** [comment]
The main point was to move out $sourceUrls from __constructor(), because this parameter is optional. Please set default value directly to class property. Setter mut be called from DI, not constructor.

**Comment by gerdemann** [comment]
> The main point was to move out $sourceUrls from __constructor(), because this parameter is optional. Please set default value directly to class property. Setter mut be called from DI, not constructor.

I have orientated myself on the secret property. Unfortunately, I'm not that deep in the project. Can you describe what you mean in more detail?

**Comment by ramunasd** [comment]
I think I was quite clear :)
Nevermind, I will fix this later.

**Comment by gerdemann** [comment]
Thank you very much.
I had also orientated myself on the Gitlab webhook, where it was done in the same way.

**Comment by ramunasd** [comment]
Rebased and merged from other branch.

**Pull request closed without merging**